controllers/clusters/cassandra_controller.go

```diff
@@ -400,30 +316,122 @@ func (r *CassandraReconciler) handleUpdateCluster(
 		return reconcile.Result{}, err
 	}
 
-	iCassandra, err := c.FromInstAPI(iData)
+	iCassandra := v1beta1.Cassandra{}
+	iCassandra.FromInstAPI(instaModel)
+
+	bootstrap := newOnPremisesBootstrap(
+		r.Client,
+		c,
+		r.EventRecorder,
+		iCassandra.Status.ToOnPremises(),
+		c.Spec.OnPremisesSpec,
+		newExposePorts(c.GetExposePorts()),
+		c.GetHeadlessPorts(),
+		c.Spec.PrivateNetwork,
+	)
+
+	ctx := context.Background()
+
+	err = handleCreateOnPremisesClusterResources(ctx, bootstrap)
```

Comment:
You can just put ctx as an func argument and use it

Reply:
Fixed